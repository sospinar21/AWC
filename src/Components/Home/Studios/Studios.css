.img-box {
  height: 150px;
  overflow: hidden;
  width: 12vw;
  border-radius: 5px;
  box-shadow: 1px 2px 3px #ccc; }

.suggStudio-cards {
  width: 90%;
  cursor: pointer;
  margin: 20px;
  padding: 20px;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  border-bottom: 1px solid #cccccc87;
  margin: 20px auto; }
  .suggStudio-cards .studio-img {
    width: 100%;
    height: auto; }
  .suggStudio-cards h4 {
    font-size: 24px;
    color: black; }
  .suggStudio-cards .text-container {
    line-height: 2;
    padding: 0 50px;
    box-sizing: border-box;
    text-align: left; }
    .suggStudio-cards .text-container .material-icons {
      font-family: 'Material Icons';
      font-weight: normal;
      font-style: normal;
      font-size: 20px;
      display: inline-block;
      line-height: 1;
      text-transform: none;
      letter-spacing: normal;
      word-wrap: normal;
      white-space: nowrap;
      direction: ltr;
      color: #b62d4e;
      line-height: 50px; }
  .suggStudio-cards b {
    font-size: 30px;
    margin: -3px 10px; }

.suggStudio-cards:hover {
  background-color: #d3d3d33b; }

.studiocards-container {
  display: flex;
  flex-wrap: wrap;
  position: relative; }

.featured-studio-text {
  background-color: rgba(255, 255, 255, 0.7);
  position: absolute;
  bottom: 50px;
  padding: 35px;
  border-radius: 3px; }
  .featured-studio-text h1, .featured-studio-text h2, .featured-studio-text p {
    color: #444; }
  .featured-studio-text h1 {
    font-size: 40px;
    color: #222b37;
    text-transform: uppercase;
    font-weight: bold; }
  .featured-studio-text p {
    margin-top: 10px;
    font-size: 1.4rem; }

.selected-card {
  display: flex;
  width: 100%;
  position: relative; }

.selected-studio-box {
  background-size: cover;
  background-position-x: center;
  background-position-y: center;
  background-repeat: no-repeat;
  background-repeat: no-repeat;
  position: relative;
  height: 355px;
  width: 100%;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0px 0px 3px lightgrey;
  background-image: url("http://www.semo.edu/images/dance-modernBanner.jpg"); }

.studios {
  width: 100%;
  height: 500px;
  overflow-y: scroll; }

.description-container {
  display: flex; }

.main-info {
  display: flex;
  width: 80.5%; }

.description {
  font-size: 20px;
  overflow-y: hidden;
  line-height: 2;
  color: #b62d4c;
  margin-top: 10px; }
  .description p {
    color: #b62d4c;
    font-size: 1rem;
    line-height: 1.6rem; }

.phone {
  color: pink; }

.b-description {
  width: 60%;
  background-color: black; }

.description-box {
  width: 300px;
  position: absolute;
  height: 1;
  padding: 20px 40px;
  bottom: 17px;
  border-radius: 3px;
  -webkit-box-shadow: 1px 2px 3px #eee;
  box-shadow: 1px 2px 3px #eee;
  margin-left: 20px;
  background: white; }
  .description-box h4 {
    font-size: 30px;
    color: black; }
  .description-box b {
    font-size: 30px;
    margin: 0 10px; }
