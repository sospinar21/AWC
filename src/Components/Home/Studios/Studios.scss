.img-box {
  height: 140px;
  overflow: hidden;
  width: 12vw;
  border-radius: 5px;
  box-shadow: 1px 2px 3px #ccc;
}

.suggStudio-cards {
  width: 90%;
  cursor: pointer;
  margin: 20px;
  padding: 20px;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  border-bottom: 1px solid #cccccc87;
  margin: 20px auto;

  .studio-img {
    width: 100%;
    height: auto;
  }

  h4 {
    font-size: 24px;
    color: black;
  }

  p{

  }

  .text-container {
    line-height: 2;
    padding: 0 50px;
    box-sizing: border-box;
    text-align: left;

    .material-icons{
      font-family: 'Material Icons';
      font-weight: normal;
      font-style: normal;
      font-size: 20px;
      display: inline-block;
      line-height: 1;
      text-transform: none;
      letter-spacing: normal;
      word-wrap: normal;
      white-space: nowrap;
      direction: ltr;
      color: #b62d4e;
      line-height: 50px;
    }
  }

  b{
    font-size: 30px;
    margin: -3px 10px;
  } 
}

.suggStudio-cards:hover{
  background-color: #d3d3d33b
 }

.studiocards-container{
  display: flex;
  flex-wrap: wrap;
  position: relative;
}

.featured-studio-text {
  background-color: rgba(255, 255, 255, 0.7);
  position: absolute;
  bottom: 50px;
  padding: 35px;
  border-radius: 3px;

  h1,h2,p {
    color: #444;
  }

  h1{
    font-size: 40px;
    color: #222b37;
    text-transform: uppercase;
    font-weight: bold;
  }

  p {
    margin-top: 10px;
    font-size: 1.4rem;
  }


}

.selected-card {
  display: flex;
  width: 100%;
  position: relative;
}

.selected-studio-box {
  background-size: cover;
  background-position-x: center;
  background-position-y: center;
  background-repeat: no-repeat;
  background-repeat: no-repeat;
  position: relative;
  height: 355px;
  width: 100%;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0px 0px 3px lightgrey;
  background-image: url('http://www.semo.edu/images/dance-modernBanner.jpg') 
}

.studios {
  width: 100%;
  height: 500px;
  overflow-y: scroll;
}

.description-container {
  display: flex
}

.main-info {
  display: flex;
  width: 80.5%;
}

.description {
  font-size: 20px;
  overflow-y: hidden;
  line-height: 2;
  color: #b62d4c;
  margin-top: 10px;

  p{
    color: #b62d4c;
    font-size: 1rem;
    line-height: 1.6rem;
  }
}

.phone {
  color: pink
}

.b-description{
  width: 60%;
  background-color: black
}

.description-box {
  width: 300px;
  position: absolute;
  height: 1;
  padding: 20px 40px;
  bottom: 17px;
  border-radius: 3px;
  -webkit-box-shadow: 1px 2px 3px #eee;
  box-shadow: 1px 2px 3px #eee;
  margin-left: 20px;
  background: rgba(255, 255, 255, 1);

  h4 {
    font-size: 30px;
    color: black;
  }

  b {
    font-size: 30px;
    margin: 0 10px;
  }
}