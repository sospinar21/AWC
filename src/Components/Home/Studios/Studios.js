import React, { Component } from 'react';
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls';
import { connect } from 'react-redux';
import { addStudios, addSelectedStudio } from '../../../Actions/actions';
import './Studios.css';
import MyMapComponent from '../Map/Map';
import PropTypes from 'prop-types';


export class Studios extends Component {
  constructor() {
    super();
    this.state = ({
      selectedStudio:false
    });
  }

  componentDidMount() {
    const city = this.props.selectedLocation;
    const selectedLocation = city.length ? city : 'usa';
    this.fetchStudios(selectedLocation);
  }

fetchStudios = async (city) => {
  let api = new ApiCalls();
  const suggestedStudios = await api.fetchStudios(city);
  await this.props.addStudios(suggestedStudios);
}

fetchSelected = (studio) => {
  let api = new ApiCalls();
  if (studio) {
    const reviews = api.fetchSingleStudio(studio.id)
      .then(response => this.props.addSelectedStudio({...response, ...studio})); 
  } else {
    return (
      <h1>Select a Studio</h1>
    );
  }
}

displaySelected = () => {
  const selected = this.props.selectedStudio;
  const review = selected.rating * 20
  if (selected.id) { 
    return (
      <div key={1} className='selected-card'> 
        <div className='map'>
          <MyMapComponent />
        </div> 
        <div className='description-box'>    
          <div className='description-container'>
            <div className='text-container'>
              <h4>{selected.title}</h4>
              <div className="stars">
                <div>
                  <img src={require('../../../assets/regular_0.png')} />
                </div>
                <div className="redStars" style={{ "width": `${review}%` }}>
                  <img src={require('../../../assets/regular_5@3x.png')} />
                </div>
              </div>
            </div>
          </div> 
          <div className='description'>
            <p>{selected.location.display_address[0]}</p>
            <p>{selected.location.display_address[1]}</p>          
            <p className='phone'>{selected.phone}</p>              
          </div>
        </div>
      </div>
    );
  }
}

displayStudios = () => {
  const studios = this.props.suggestedStudios.map(studio => {
    return (
      <div className="post" key={studio.id} onClick={() => this.fetchSelected(studio)}>
        <div className="poster poster-studio">
          <b className='event-b'>{studio.title}</b>
          <div className='description'>
            <p>{studio.location.display_address[0]}</p>
            <p>{studio.location.display_address[1]}</p>          
            <p className='phone'>{studio.phone}</p>              
          </div>
          <br/>
        </div>
        <div className="content content-s">
          <img className='studio-img' src={studio.image} />
        </div>
      </div> 
    );
  });

  return studios;
}


render () {  
  if (this.props.suggestedStudios.length) {
    return (
      <div className='studiocards-container'>
        <div className='selected-studio-box'>
          <div className='featured-studio-text'>
            <h1>Studios</h1>
            <h2>Find studios around the world</h2>
            <p>click below!</p>
          </div>
          {this.displaySelected()}
        </div>
        <div className ='studios'>
          {this.displayStudios()}
        </div>
      </div>
    );

  } else {
    return (
      <div className='loading'>
        <h1 >Loading...</h1>
      </div>
    );
  }
}
}

export const mapStateToProps = (state) => {
  return ({
    suggestedStudios: state.suggestedStudios,
    selectedLocation: state.location,
    selectedStudio: state.selectedStudio
  });
};

export const mapDispatchToProps = dispatch => ({
  addStudios: (studiosData) => dispatch(addStudios(studiosData)),
  addSelectedStudio: (selected) => dispatch(addSelectedStudio(selected))
});

Studios.propTypes = {
  suggestedStudios: PropTypes.object,
  addStudios: PropTypes.func,
  addSelectedStudio: PropTypes.func,
  selectedStudio: PropTypes.object,
  selectedLocation: PropTypes.string
};

export default connect(mapStateToProps, mapDispatchToProps)(Studios);