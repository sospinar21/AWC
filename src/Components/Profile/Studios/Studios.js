import React, { Component } from 'react'
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls'
import { connect } from 'react-redux';
// import 
import './Studios.css'

export class Studios extends Component {
  



  render () {
    return(
      <div>

      </div>
    )
  }
}

// export const mapStateToProps = {}

// export const mapDispatchToProps = () => ({})

// export default connect(mapStateToProps, mapDispatchToProps)(Studios)