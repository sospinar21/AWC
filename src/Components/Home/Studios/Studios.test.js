import {Studios, mapStateToProps, mapDispatchToProps} from '../Studios/Studios';
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';


describe('Studios', () => {

  let studios;
  let mockProps;

  beforeEach(() => {

    mockProps = {
      suggestedStudios: [{name: 'party',
      location:'denver'
    }],
      addStudios: jest.fn()
    };

    studios = shallow(<Studios {...mockProps}/>, { disableLifecycleMethods: true });

  });

  it('matches snapshot', () => {
    expect(studios).toMatchSnapshot();
  });
});