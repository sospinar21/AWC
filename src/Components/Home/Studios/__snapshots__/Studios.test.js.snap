// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Studios matches snapshot 1`] = `undefined`;
